contracts/BucketAuction.sol

```diff
@@ -11,10 +11,13 @@ import "./IBucketAuction.sol";
 import "./ERC721M.sol";
 
 contract BucketAuction is IBucketAuction, ERC721M {
+    
+    uint256 private _startTimeUnixSeconds;
+    uint256 private _endTimeUnixSeconds;
     bool private _claimable;
     uint256 private _minimumContributionInWei;
     uint256 private _price;
-    bool private _auctionActive;
+    bool private _firstTokenSent;
```

Comment:
nit: we can group the `bool` variables together at the beginning to save 1 storage slot:

```
bool private _claimable;
bool private _firstTokenSent;
// rest of the variables
```

This is because each storage slot is 32 bytes and fixed size types (bool, uint, address) can be packed into a single storage slot as long as they fit: https://fravoll.github.io/solidity-patterns/tight_variable_packing.html